### ISSUE TITLE
Upgrading from 11.2.2 to 11.3.0 breaks projections

### ISSUE THREAD

**Issue by RaYell** [ChilliCream/hotchocolate#3820]
**Describe the bug**
Projections do not seem to work correctly in 11.3.0. The `IsProjected()` call on the property seems to be ignored.

**To Reproduce**
1. Update HC to 11.3.0
2. Call a query using type with a property decorated with `IsProjected()` but do not include this property in your request
3. Check the generated EF query (the projected column is not there)

**Expected behavior**
I expect projections to work exactly the same between 11.2 and 11.3 releases (according to semver there should not be any breaking changes)

**Additional context**
None

**Comment by michaelstaib** [comment]
@PascalSenn can you verify.

**Comment by arif-hanif** [comment]
@RaYell are you using Paging?

**Comment by RaYell** [comment]
Yes, I am. Paging, projection, filtering and sorting in that order.

**Comment by tobias-tengler** [comment]
I tried to reproduce this issue [here](https://github.com/tobias-tengler/hc-issue-3820), but failed to do so - everything works as expected.
Could you provide a minimal repro that showcases the issue you are having?

**Comment by RaYell** [comment]
I've dig deeper into my app after upgrade to v11.3 and I here's what I found.

If I have an object with multiple fields marked as `IsProjected` then only the last field is projected, and all of the others are ignored.

Here's an example

I have a type that looks more less like this.
```csharp
public class ContactType : ObjectType<Contact>
{
    protected override void Configure(IObjectTypeDescriptor<Contact> descriptor)
    {
        descriptor.Field(x => x.ContactNo).IsProjected();
        descriptor.Field(x => x.CustomerNo).IsProjected();
        descriptor.Field(x => x.LanguageCode).IsProjected();

        // ...
    }
}
```

and I register it like this
```csharp
descriptor.Field(FieldContacts)
    .Type<ListType<ContactType>>()
    .Resolve(ctx => ctx.Service<MyContext>().Contacts)
    .UsePaging<ContactType>()
    .UseProjection()
    .UseFiltering<ContactFilterType>()
    .UseSorting<ContactSortType>();
```


If I run the simplest query to retrieve it
```
{
  contacts(first: 10) {
    nodes {
      name
    }
  }
}
```

I see that EF is generating query like this one
```sql
SELECT TOP(@__p_0) [c].[Name], [c].[LanguageCode]
      FROM [dbo].[Contacts] AS [c]
```

As you can see it only used the last property defined as projected and ignored the rest. Now if I comment out the line where `LanguageCode` is marked as projected and run the same query again I see EF generates query like this
```sql
SELECT TOP(@__p_0) [c].[Name], [c].[CustomerNo]
      FROM [dbo].[Contacts] AS [c]
```

In 11.2.* this is working fine and always all of the projections are resolved with the query, but in 11.3 it seems to take only the last one into account.

**Comment by tobias-tengler** [comment]
@RaYell Thanks a bunch for the details, I'll try to reproduce it later today as well. Those are some good pointers!

**Comment by tobias-tengler** [comment]
I managed to reproduce this behavior both with 11.3.0 and the latest preview of 12.

**Comment by michaelstaib** [comment]
@PascalSenn is this fixed? and if in which version?

**Comment by RaYell** [comment]
Looking at the release notes I think 11.3.2 has this issue fixed.

**Comment by michaelstaib** [comment]
Closing this issue as it has ben fixed with 11.3.2
